### ISSUE TITLE
Set up email alert subscriber system

### ISSUE THREAD

**Issue by anderspeders** [NRGI/resourcecontracts.org#126]
As a subscriber I want an alert when - a contract: 
- from X country is uploaded
- from X corporate group is uploaded

**Comment by manishgs** [comment]
**Design/Workflow on Frontend.**

1) There will be subscription form at the footer of the page.

![1 footer subscription btn](https://cloud.githubusercontent.com/assets/3942999/20204753/8b6ab2c2-a7fa-11e6-8014-7f90931ddb06.png)

2) When user submit the form, a popup window will be shown as below:

![2 subscription detail](https://cloud.githubusercontent.com/assets/3942999/20204758/9c6d339c-a7fa-11e6-8228-09a44a0a787b.png)

user have option to subscribe for all countries or specific country, Likewise to subscribe for all corporate group or a single corporate group.

3) When user submit the form thank you message will be shown as below.

![3 thank you](https://cloud.githubusercontent.com/assets/3942999/20204777/bd662702-a7fa-11e6-8c28-f14e939242c6.png)


**Backend workflow**

1) Separate system will be build to handle subscription system.
subscribe.resourcecontract.org (proposed) subdomain will be created/used for subscription.

2) All the subscribers' information will be store/view in the above subdomain.

3) When a contract is published on the admin for thee first time, a notification will be sent to all subscribed users. Later If contract is unpublished and then re-published, notification will not be sent.

4) There will be unsubscribe link at the button of the email template where user can unsubscribe anytime  from the notification system.


@anderspeders @SamCCSI @charlesyoung 
Let us know your comments and suggestions.

**Comment by charlesyoung** [comment]
Thanks @manishgs 

Will another notification be sent out for contracts (that already have had the metadata published) that are annotated?

**Comment by manishgs** [comment]
No, system will not send notification when pdf text or annotation is published. Notification will be sent when contract metadata is published.

We can discuss if we need the feature to send notification on pdf text and annotation publish.

**Comment by charlesyoung** [comment]
@KaitlinCCSI @SamCCSI @anderspeders do we only send a notification when the contract metadata is published or also when the PDF text or annotation is published?

**Comment by charlesyoung** [comment]
@anjesh no feedback received so please implement as suggested. We can always extend the subscriber system later on if required.

**Comment by charlesyoung** [comment]
@anjesh can you set this up on the demo site?

**Comment by charlesyoung** [comment]
@anderspeders can we also include company name, additional to country and corporate group, because only a small proportion of contracts have corporate group captured?

**Comment by charlesyoung** [comment]
@anderspeders @SamCCSI @KaitlinCCSI can we also include company name, additional to country and corporate group, because only a small proportion of contracts have corporate group captured?

This function has been developed so just waiting for feedback on above. 

cc @manishgs

**Comment by anderspeders** [comment]
Sounds good to me. If this does not include a substantial amount of extra work.

**Comment by KaitlinCCSI** [comment]
Fine with me too, though I wonder how many situations this will actually be relevant for -- how many times will a company already be in system (for someone to find in the dropdown list) with then new contracts from same company, or updates to existing contracts of that company, made? 

On separate note, the thank you message should probably be refined; assume @jedm might have some suggestions?

**Comment by jedm** [comment]
I agree that the use case for this as
spec-ed will be limited. 

Suggest that the most likely need will be on single contract pages, country pages and search result pages when user searches single country or resource, e.g.,

- notify me when this country has a new contract
- notify me when this resource has a new contract 
- notify me when this document has annotations (for unannotated docs)

YI, 1) are all three of these scenarios possible, and 2) is there a way to offer notification signup from the contextual pages instead of only from homepage?

**Comment by manishgs** [comment]
@jedm 

> YI, 1) are all three of these scenarios possible,

1, 2 is possible 
for third, it is possible when annotation is published for first time. if annotation added later and publish then notification will not be sent.

> 2) is there a way to offer notification signup from the contextual pages instead of only from homepage?

signup form will be present at the bottom of all pages.

**Comment by shashikhanal** [comment]
Newsletter system workflow for testing:
1. Go to any [subsites](http://rc-site-demo.elasticbeanstalk.com/rc/public/) and fill up the subscription field and click subscribe button.
![subsite-homepage](https://cloud.githubusercontent.com/assets/4457892/21562158/5ff8ae30-ce9e-11e6-95eb-93084ae2a838.png)
2. Now you reach [newsletter system](http://rc-site-demo.elasticbeanstalk.com/newsletter/public/). Here fill up the desired fields and click the subscribe button.
![newsletter-homepage](https://cloud.githubusercontent.com/assets/4457892/21562526/981b8d0c-cea1-11e6-996b-8a4586f75c13.png)
3. Now you get an acknowledgement page and receive a confirmation mail.
![thanks-subscribe](https://cloud.githubusercontent.com/assets/4457892/21562630/832ca060-cea2-11e6-9afc-3a10f4755920.png)
![confirmation-email](https://cloud.githubusercontent.com/assets/4457892/21562658/b0835ea0-cea2-11e6-981e-54391cffbdc2.png)
4. Click "Click to confirm your email." button. Now you get a confirmation message.
![confirmation-message](https://cloud.githubusercontent.com/assets/4457892/21562692/fdc490b2-cea2-11e6-9f83-f1976960cf05.png)
5. Now go to [admin section](http://rc-site-demo.elasticbeanstalk.com/newsletter/public/login) and login with correct admin username and password.
![login-page](https://cloud.githubusercontent.com/assets/4457892/21562963/3640c0c6-cea5-11e6-8729-764acb7ee300.png)
6. Now your dashboard looks as follows:
![dashboard-looks](https://cloud.githubusercontent.com/assets/4457892/21562721/3da295a8-cea3-11e6-8d02-0d6796b9fbc8.png)
7. Contracts page looks as follows:
![contracts-page](https://cloud.githubusercontent.com/assets/4457892/21562740/68625314-cea3-11e6-960a-688c1afcfe83.png)
8. Settings page looks as follows. Now in order to get subscription email set the 'Time' field in UTC timezone accordingly.
![settings-page](https://cloud.githubusercontent.com/assets/4457892/21562774/a041c242-cea3-11e6-9b68-a9a5a8561e7d.png)
9. You can logout clicking the logout button.
![logout-red](https://cloud.githubusercontent.com/assets/4457892/21562869/75542a88-cea4-11e6-887c-8957fc64e735.png)
10. Subscription email looks as follows:
![subscription-email](https://cloud.githubusercontent.com/assets/4457892/21563144/e0cfb4d8-cea6-11e6-9345-4f4425e12afd.png)

**Comment by jedm** [comment]
@shashikhanal and @manishgs thank you for these examples and documentation.

Can you clarify if user can subscribe to a specific contract **directly from the contract page?**

ie, can you create a "doc-specific" subscription link higher up on the UI? 

If not, I am not sure that the usage will be sufficient enough to justify the coding time.

Meanwhile, wishing you all a very Happy New Year! Looking forward to doing more in 2017.

Jed

**Comment by manishgs** [comment]
@jedm  Thank you and wish you the same.

> Can you clarify if user can subscribe to a specific contract directly from the contract page?
Current scope is to subscribe from the system not to a specific contract. But we can discuss for your suggestion. 

Could please provide example why user subscribes to specific contract and what type of updates should notify to subscriber?

**Comment by jedm** [comment]
Sorry if I was not clear, the specific contract subscription is for when annotations are added to a specific contract (for the first time). Per previous question here: https://github.com/NRGI/resourcecontracts.org/issues/126#issuecomment-268067687

The other question was regarding "contextual subscriptions." User should be able to click from a specific page (country, resource, contract) directly to a signup for notifications regarding THAT page/list/contract. Same question as here: https://github.com/NRGI/resourcecontracts.org/issues/126#issuecomment-269746723 

Thanks.

**Comment by charlesyoung** [comment]
Putting this ticket on-hold for now to be discussed in more detail during the 20/01 call.

**Comment by charlesyoung** [comment]
Ticket not prioritized. When @jedm conducts use case research he could look into user demand for this.

**Comment by charlesyoung** [comment]
Something to reconsider again? 

Below is how ResourceProjects allows for alerts/notifications. 

https://resourceprojects.org/subscribe

**Comment by man-git007** [comment]
@charlesyoung I think we need discussion for this ticket.

cc @jedm
